qcodes/dataset/data_set.py

```diff
@@ -667,8 +709,15 @@ def mark_completed(self) -> None:
             raise RuntimeError('Can not mark DataSet as complete before it '
                                'has been marked as started.')
         self.completed = True
+        self._perform_completion_actions()
```

Comment:
Logically it seems like these two lines should be reordered. Not sure that it matters

Reply:
Good point!